apps/sr-frontend/src/app/Character.tsx

```diff
@@ -50,11 +65,129 @@ export default function Character() {
               inputProps={{ min: 0, max: 6 }}
               value={character?.ascension || 0}
               onChange={(e) =>
-                charReducer({
+                updateCharacter({
                   ascension: parseInt(e.target.value) as AscensionKey,
                 })
               }
+              disabled={!character}
             />
+            <Grid container>
+              <Grid item>
+                <DropdownButton
+                  title={`Eidolon Lv. ${character?.eidolon ?? 0}`}
+                  fullWidth={false}
+                  disabled={!character}
+                >
+                  {allEidolonKeys.map((eidolon) => (
+                    <MenuItem
+                      key={eidolon}
+                      selected={character?.eidolon === eidolon}
+                      disabled={character?.eidolon === eidolon}
+                      onClick={() => updateCharacter({ eidolon })}
+                    >
+                      Eidolon Lv. {eidolon}
+                    </MenuItem>
+                  ))}
+                </DropdownButton>
+              </Grid>
+            </Grid>
+            <Accordion>
+              <AccordionSummary expandIcon={<ExpandMoreIcon />}>
+                Traces
+              </AccordionSummary>
+              <AccordionDetails>
+                <Stack spacing={1}>
+                  <Typography variant="h6">Abilites</Typography>
+                  <Divider />
+                  <Grid container gap={1}>
+                    <Grid item>
+                      <AbilityDropdown
+                        characterKey={characterKey}
+                        abilityKey="basic"
+                      />
+                    </Grid>
+                    <Grid item>
+                      <AbilityDropdown
+                        characterKey={characterKey}
+                        abilityKey="skill"
+                      />
+                    </Grid>
+                    <Grid item>
+                      <AbilityDropdown
+                        characterKey={characterKey}
+                        abilityKey="ult"
+                      />
+                    </Grid>
+                    <Grid item>
+                      <AbilityDropdown
+                        characterKey={characterKey}
+                        abilityKey="talent"
+                      />
+                    </Grid>
+                  </Grid>
+                  <Typography variant="h6">Bonus Abilities</Typography>
+                  <Divider />
+                  <Grid container gap={1}>
+                    {Object.entries(character?.bonusAbilities ?? {}).map(
+                      ([bonusAbility, enabled]) => (
+                        <Grid item key={bonusAbility}>
+                          <Button
+                            color="primary"
```

Comment:
Sorry missed this earlier. Can you make it so the color is `success` when checked?